cardano-node/src/Cardano/CLI/Run.hs

```diff
@@ -356,64 +377,113 @@ runCommand CLIOps{..}
   unless (null issues) $
     throwIO $ CertificateValidationErrors certF issues
 
-runCommand CLIOps{..}
-           (SubmitTx stTopology (TxFile stTx) stCommon) = do
-
-  cc <- mkConfiguration mainnetConfiguration stCommon
-
-  SomeProtocol p <- fromProtocol cc coProtocol
-
-  case p of
-    ProtocolRealPBFT{} -> do
+runCommand co (SubmitTx stTopology (TxFile stTx) stCommon) = do
+  withRealPBFT co mainnetConfiguration stCommon $
+    \cc p@(ProtocolRealPBFT _ _ _ _ _) -> do
       txBS <- LB.readFile stTx
       case deserialiseOrFail txBS of
         Left  e  -> throwIO $ TxDeserialisationFailed stTx e
         Right tx@ByronTx{byronTxId} -> do
           putStrLn $ "transaction hash (TxId): " <> show byronTxId
           handleTxSubmission cc p stTopology tx stdoutTracer
-    _ -> throwIO $ ProtocolNotSupported coProtocol
 
 runCommand co@CLIOps{..}
            (SpendGenesisUTxO
             (NewTxFile ctTx) ctKey ctGenRichAddr ctOuts ctCommon) = do
-
-  sk <- readSigningKey co ctKey
-
-  cc <- mkConfiguration mainnetConfiguration ctCommon
-
-  SomeProtocol p <- fromProtocol cc coProtocol
-
-  case p of
-    ProtocolRealPBFT gc _ _ _ _ -> do
+  withRealPBFT co mainnetConfiguration ctCommon $
+    \_cc (ProtocolRealPBFT gc _ _ _ _) -> do
+      sk <- readSigningKey co ctKey
       let tx@ByronTx{byronTxId} = txSpendGenesisUTxOByronPBFT gc sk ctGenRichAddr ctOuts
       putStrLn $ "genesis protocol magic:  " <> show (configProtocolMagicId gc)
       putStrLn $ "transaction hash (TxId): " <> show byronTxId
       ensureNewFileLBS ctTx (serialise tx)
-    _ -> throwIO $ ProtocolNotSupported coProtocol
+
+runCommand co@CLIOps{..}
+           (SpendUTxO
+            (NewTxFile ctTx) ctKey ctIns ctOuts ctCommon) = do
+  withRealPBFT co mainnetConfiguration ctCommon $
+    \_cc (ProtocolRealPBFT gc _ _ _ _) -> do
+      sk <- readSigningKey co ctKey
+      let tx@ByronTx{byronTxId} = txSpendUTxOByronPBFT gc sk ctIns ctOuts
+      putStrLn $ "genesis protocol magic:  " <> show (configProtocolMagicId gc)
+      putStrLn $ "transaction hash (TxId): " <> show byronTxId
+      ensureNewFileLBS ctTx (serialise tx)
 
 {-------------------------------------------------------------------------------
   Supporting functions
 -------------------------------------------------------------------------------}
 
-txSpendGenesisUTxOByronPBFT :: CC.Genesis.Config -> SigningKey -> Address -> NonEmpty TxOut -> GenTx (ByronBlockOrEBB ByronConfig)
-txSpendGenesisUTxOByronPBFT gc sk genRichAddr outs =
-  let vk         = Crypto.toVerification sk
-      txattrs    = mkAttributes ()
-      tx         = UnsafeTx (pure txIn) outs txattrs
-      txIn      :: CC.UTxO.TxIn
-      txIn       = handleMissingAddr $ fst <$> Map.lookup genRichAddr initialUtxo
+withRealPBFT
+  :: CLIOps IO
+  -> PartialCardanoConfiguration
+  -> CommonCLI
+  -> (RunDemo (ByronBlockOrEBB ByronConfig)
+      => CardanoConfiguration
```

Comment:
It's difficult for me to infer what is happening here. Can you document with comments on what `withRealPBFT` is doing and perhaps wrap 
```
(RunDemo (ByronBlockOrEBB ByronConfig)
      => CardanoConfiguration
      -> Consensus.Protocol (ByronBlockOrEBB ByronConfig)
      -> IO ())
``` 
in a `newtype` and also annotate with comments?

Reply:
@intricate any suggestions?

Reply:
I wouldn't agree with the `newtype` point but adding a quick one-liner of doc for `withRealPBFT` would be nice.

Reply:
Done!